server/src/test/java/org/apache/druid/server/coordinator/rules/LoadRuleTest.java

```diff
@@ -685,6 +685,210 @@ public void testMaxLoadingQueueSize()
     EasyMock.verify(throttler, mockBalancerStrategy);
   }
 
+  /**
+   * 2 servers in different tiers, the first is in maitenance mode.
+   * Should not load a segment to the server in maintenance mode.
+   */
+  @Test
+  public void testLoadDuringMaitenance()
+  {
+    final LoadQueuePeon mockPeon1 = createEmptyPeon();
+    final LoadQueuePeon mockPeon2 = createOneCallPeonMock();
+
+    LoadRule rule = createLoadRule(ImmutableMap.of(
+        "tier1", 1,
+        "tier2", 1
+    ));
+
+    final DataSegment segment = createDataSegment("foo");
+
+    EasyMock.expect(mockBalancerStrategy.findNewSegmentHomeReplicator(EasyMock.anyObject(), EasyMock.anyObject()))
+            .andDelegateTo(balancerStrategy)
+            .times(1);
+
+    EasyMock.replay(mockPeon1, mockPeon2, mockBalancerStrategy);
+
+
+    DruidCluster druidCluster = new DruidCluster(
+        null,
+        ImmutableMap.of(
+            "tier1",
+            Stream.of(
```

Comment:
For testing purposes, it would be easier to have `Map<String, Iterable<ServerHolder>>` parameter in the exposed `DruidCluster` constructor, and convert to TreeSets internally. It's tedious to provide the proper comparator everywhere in tests, for example here you didn't do it.

Reply:
The order of servers doesn't matter in my tests (ie the cost-based strategy goes through all the servers independently and segments move quota is large enough to accommodate all) and I tried to make as less refactoring of existing code as possible for brevity.

Reply:
It breaks the contract of `DruidCluster`, in a way. Benignantly at the moment, but it could break. Constructors shouldn't just accept a collection of collections as a parameter and assign to a field without defensive copy / invariant check. Performance is not an issue, since this constructor is used only in tests.

Reply:
hmm, there is a complication, one test uses its own comparator org/apache/druid/server/coordinator/DruidCoordinatorBalancerTest.java:520

Reply:
btw, those tests run successfully no matter which comparator is used

Reply:
I need to review those tests before intruducing the new constructor.